rtlib/include/rt/Data.hpp

```diff
@@ -0,0 +1,71 @@
+//
+// Created by Anthony Tamasi on 2019-10-17.
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <boost/filesystem.hpp>
+#include <opencv2/core.hpp>
+
+#include "rt/ImageTypes.hpp"
+#include "rt/types/ITKMesh.hpp"
+
+namespace et = envitools;
+namespace fs = boost::filesystem;
+
+
+namespace Data {
+    class Data
+    {
+        using Pointer = std::shared_ptr<Data>; //DOES THIS GO HERE?
+
+    public:
+        static Pointer Load(const fs::path& path);
+        virtual cv::Mat getImage(int idx = 0) = 0;
+
+    protected:
+        Data() = default;
+        Data(std::string path) {} //Constructor
```

Comment:
`fs::path` and all functions should be virtual with no implementation

Reply:
What other functions need to be virtual? I thought Load() wasn't going to be virtual because it was that switch statement?